pkg/action/action.go

```diff
@@ -0,0 +1,101 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/libopenstorage/stork/drivers/volume"
+	storkv1 "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/libopenstorage/stork/pkg/controllers"
+	"github.com/libopenstorage/stork/pkg/k8sutils"
+	"github.com/portworx/sched-ops/k8s/apiextensions"
+	"github.com/sirupsen/logrus"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/record"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	validateCRDInterval time.Duration = 5 * time.Second
+	validateCRDTimeout  time.Duration = 1 * time.Minute
+)
+
+func NewActionController(mgr manager.Manager, d volume.Driver, r record.EventRecorder) *ActionController {
+	return &ActionController{
+		client:    mgr.GetClient(),
+		volDriver: d,
+		recorder:  r,
+	}
+}
+
+type ActionController struct {
+	client    runtimeclient.Client
+	volDriver volume.Driver
+	recorder  record.EventRecorder
+}
+
+func (ac *ActionController) Init(mgr manager.Manager) error {
+	err := ac.createCRD()
+	if err != nil {
+		return err
+	}
+	return controllers.RegisterTo(mgr, "action-controller", ac, &storkv1.Action{})
+}
+
+func (ac *ActionController) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	action := &storkv1.Action{}
+	err := ac.client.Get(context.TODO(), request.NamespacedName, action)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{RequeueAfter: controllers.DefaultRequeueError}, err
+	}
+
+	if action.Status == storkv1.ActionStatusScheduled {
+		if err = ac.handle(context.TODO(), action); err != nil {
+			logrus.Errorf("%s: %s: %s", reflect.TypeOf(ac), action.Name, err)
+			return reconcile.Result{RequeueAfter: controllers.DefaultRequeueError}, err
+		}
+	}
+
+	return reconcile.Result{RequeueAfter: controllers.DefaultRequeue}, nil
+}
+
+func (ac *ActionController) handle(ctx context.Context, action *storkv1.Action) error {
+	ac.updateStatus(action, storkv1.ActionStatusInProgress)
```

Comment:
What is the point of updating the status to InProgress if we are anyways updating it to Successful immediately on line 75? 

I guess it is fine for now until we add the actual logic to check the status as well and then based on status perform an operation.